src/drivers/tty/serial/ttys_processing.c

```diff
@@ -41,23 +41,30 @@ static inline struct uart *tty2uart(struct tty *tty) {
 	return tu->uart;
 }
 
-static int uart_rx_buff_put(struct uart *dev, int c) {
+static int uart_rx_buff_put(unsigned int irq_nr, struct uart *dev) {
 	struct uart_rx *rx;
 
-	irq_lock();
-	{
-		rx = pool_alloc(&uart_rx_buff);
-		if (!rx) {
+	if (dev->tty) {
+		while (uart_hasrx(dev)) {
+			irq_lock();
+			{
+				rx = pool_alloc(&uart_rx_buff);
+				if (!rx) {
+					dev->uart_ops->uart_irq_dis(dev, &dev->params);
```

Comment:
You must check whether dev->uart_ops->uart_irq_dis exist or NULL!

Reply:
Okay! done.